client/src/app/site/pages/meetings/pages/interaction/modules/interaction-container/components/video-player/video-player.component.ts

```diff
@@ -56,19 +58,39 @@ export class VideoPlayerComponent implements AfterViewInit, OnDestroy {
         this._videoUrl = value.trim();
         this.playerType = this.determinePlayer(this.videoUrl);
 
+        if (!this.usingVjs) {
+            this.stopVJS();
+            this.unloadVjs();
+        }
+        if (this.nanoPlayer) {
+            this.nanoPlayer.destroy();
+        }
+
         if (this.usingVjs) {
             this.mimeType = this.determineContentTypeByUrl(this.videoUrl);
             if (this.afterViewInitDone) {
                 this.initVjs();
             }
+        } else if (this.usingNanocosmos) {
+            this.videoId = this.getNanocosmosVideoId(this.videoUrl);
+
+            if (!this.nanoPlayer) {
+                const script = document.createElement(`script`);
+                script.type = `text/javascript`;
+                script.src = `https://demo.nanocosmos.de/nanoplayer/api/release/nanoplayer.4.min.js`;
+                script.onload = () => {
+                    // @ts-ignore
```

Comment:
Why are you using ts-ignore here?
At least add a comment as to why

Reply:
Also: This function is way too long, would it be possible to move some of it to a helper function or two?

Reply:
`NanoPlayer` is loaded from an external source and therefore not known to typescript. I don't think this needs explanation as this is already pretty clear by looking at the code above which embeds the external scripts.